[Magic Network Monitoring] Simulate DDoS attack tut (#12444)
cloudflare/cloudflare-docs: content/magic-network-monitoring/tutorials/ddos-testing-guide.md

### MHDDoS
- Recommended for Windows users
- Download the source code: https://github.com/MatrixTM/MHDDoS
- Read the documentation: https://github.com/MatrixTM/MHDDoS/wiki

### Kali Linux & hping3
- Recommended for Linux users. This is an established cybersecurity tool with a variety of use cases beyond DDoS attacks.
- Download the Linux distribution: https://www.kali.org/
- Read the Kali Linux documentation: https://www.kali.org/docs/
- Learn more about the hping3 package: https://www.kali.org/tools/hping3/

For the purposes of this tutorial, we will be using the MHDDoS tool.

---

## Prerequisites

MGDDoS requires Python 3, Python 3 libraries, and Git. If you already have these tools installed, follow the [MHDDoS automatic installation guide](https://github.com/MatrixTM/MHDDoS/wiki/installation).

If you are an Enterprise customer, you also need to [open a support ticket with Cloudflare](/ddos-protection/reference/simulate-ddos-attack/) when simulating DDoS attacks against your own Internet properties.

---

## Install the tools

### 1. Install Python 3

1. Go to the [Python download page](https://www.python.org/downloads/), and download the latest version of Python.

> 2. Follow the recommended installation settings, and make sure to select the option **Add Python PATH** to add Python to your file path.

Reviewer: ```suggestion
2. Follow the recommended installation settings, and make sure to select the option **Add Python to PATH** to add Python to your file path.
```
Checked in Python 3.12.1's installer:

![image](https://github.com/cloudflare/cloudflare-docs/assets/680496/4b42f7ad-9ab7-4adf-9432-4bdefa425c44)

The `.exe` part is Windows-specific, so we may want to leave that out. However, since we'll be using Windows for the tutorial, we could use `python.exe` here.

3. Open your command prompt, and run the following command to confirm Python has been installed and added to your computer’s program path:

```bash
C:\> python --version
```

You should receive a message with the current version of Python.

### 2. Install Git

1. Go to the [Git download page](https://git-scm.com/download/win), and download the latest version of Git.
2. Follow the recommended installation settings.

> 3. Open a terminal window, and run the following command to confirm Git has been installed:

Reviewer: ```suggestion
3. Open a command prompt, and run the following command to confirm Git has been installed:
```
I'd use the same name as in "Install Python 3" (and yes, we should open a _new_ command prompt).

```bash
C:\> git --version
```

You should receive a message with the current version of Git.

### 3. Install MHDDoS

> 1. Go to the [installation guide](https://github.com/MatrixTM/MHDDoS/wiki/installation) for MHDDoS in the project’s GitHub Repository Wiki.

Reviewer: ```suggestion
1. Go to the [installation guide](https://github.com/MatrixTM/MHDDoS/wiki/installation) for MHDDoS in the project’s GitHub repository wiki.
```
I don't think we need to capitalize these two words.

2. Follow the steps for either **Automatic Installation**, **DockerFile**, or **Manual Installation** depending on your preference.

> {{<Aside type="note">}}The steps for **Automatic Installation** are a good default to follow unless you already have docker set up and are familiar with it.{{</Aside>}}

Reviewer: ```suggestion
{{<Aside type="note">}}The steps for **Automatic Installation** are a good default to follow unless you already have Docker set up and are familiar with it.{{</Aside>}}
```

### 4. Check if MHDDoS is working

The last step is to check if the MHDDoS tool is working. For that, we will simulate a small DDoS attack against 1.1.1.1.

> To review all the attack options relevant in testing Magic Network Monitoring, visit the [MHDDoS README.md](https://github.com/MatrixTM/MHDDoS#readme), and scroll down to **Features And Method > Layer 4**. To learn how to structure the commands for starting a test attack, read the [MHDDoS Usage of Script](https://github.com/MatrixTM/MHDDoS/wiki/Usage-of-script) wiki article.

Reviewer: ```suggestion
To review all the attack options relevant in testing Magic Network Monitoring, visit the [MHDDoS README](https://github.com/MatrixTM/MHDDoS#readme), and scroll down to **Features And Method > Layer 4**. To learn how to structure the commands for starting a test attack, read the [MHDDoS Usage of Script](https://github.com/MatrixTM/MHDDoS/wiki/Usage-of-script) wiki article.
```

1. Open the command prompt, and navigate to the folder `MHDDoS-main`.
2. Run the following command:

```bash

> C:\> python start.py udp 1.1.1.1:53 1 10 true

Reviewer: ```suggestion
C:\MHDDoS-main> python start.py udp 1.1.1.1:53 1 10 true
```
Since the user will be in the `MHDDoS-main` folder now.

8. Select **Save**.

Refer to the [Rules](/magic-network-monitoring/rules/) page for more information on how to set up and edit rules.

### 4. Identify the traffic protocol, IP address, and port for testing

The next step involves analyzing data received over a few weeks. This is needed so we can have a better sense of the type of traffic received and what port we need for testing.

1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/login), and select your account.
2. Go to  **Analytics & Logs** > **Magic Monitoring**.
3. Select **Add filter*.
4. In **New filter**, create the following two rules:

  ```txt
  Monitoring Rule | equals | example_192.168.1.1
  ```

  ```txt
  Protocol | equals | UDP
  ```

In our example, the fictional data for `example_192.168.1.1` over 21 days (the rule we created in [step 3](#3-create-mnm-rule-for-testing)) shows that the destination IP address `192.168.1.1` has received UDP traffic on **Destination Port** `137`. You can use this destination port for testing.

### 5. Simulate a DDoS attack to test MNM

> 1. Open your command line.
> 2. Navigate to the folder `MHDDoS-main`.

Reviewer: ```suggestion
1. Open a command prompt, and navigate to the folder `MHDDoS-main`.
```
Maybe "go to the folder"?

> 3. Run the following command. Adapt the destination IP address to the one you found that works for your particular case:

Reviewer: ```suggestion
2. Run the following command. Adapt the destination IP address to the one you found that works for your particular case:
```

```bash
C:\> python start.py udp 192.168.1:137 2 120 true
```

> This command will simulate a UDP DDoS attack on `192.168.1.1:137`. The UDP attack will use two Python threads, and run for 120 seconds. Debug mode is set to `true`, which lets us observe the attack's live progress. This command should generate anywhere from 25 to 75 mbps of attack traffic.

Reviewer: ```suggestion
This command will simulate a UDP DDoS attack on `192.168.1.1:137`. The UDP attack will use two Python threads, and run for 120 seconds. Debug mode is set to `true`, which lets us observe the attack's live progress. This command should generate anywhere from 25 to 75 Mbps of attack traffic.
```
